src/test/java/com/google/edith/UserInsightsTest.java

```diff
@@ -0,0 +1,221 @@
+package com.google.sps;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.sps.servlets.UserInsights;
+import com.google.sps.servlets.Item;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.Test;
+
+public final class UserInsightsTest {
+  private static final String USER_ID = "userId";
+  private static UserInsights userInsights;
+  private static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+  private static final LocalServiceTestHelper testHelper = 
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+    
+  @Before
+  public void setUp() {
+    userInsights = new UserInsights(USER_ID);
+    testHelper.setUp();    
+  }
+
+  @After
+  public void tearDown() {
+    testHelper.tearDown();
+  }
+
+  @Test
+  public void creatUserInisghts() {
+    // A new UserStats Entity should be added to the datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    userInsights.createUserStats();
+    Assert.assertEquals(1, datastore.prepare(new Query("UserStats")).countEntities());
```

Comment:
It seems to be using Deprecated method.
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery
can we use
```
countEntities(FetchOptions fetchOptions)
```
instead

Reply:
Done.